app/controllers/chat_controller.rb

```diff
@@ -331,15 +331,39 @@ def dismiss_retention_reminder
   def quote_messages
     params.require(:message_ids)
 
-    @chat_channel = ChatChannel.find_by(id: params[:chat_channel_id])
-    raise Discourse::NotFound if @chat_channel.blank?
-    raise Discourse::InvalidAccess if !guardian.can_see_chat_channel?(@chat_channel)
+    @chat_channel = find_channel_and_check_access(params[:chat_channel_id])
 
     message_ids = params[:message_ids].map(&:to_i)
     markdown = ChatTranscriptService.new(@chat_channel, messages_or_ids: message_ids).generate_markdown
     render json: success_json.merge(markdown: markdown)
   end
 
+  def move_messages_to_channel
+    params.require(:message_ids)
+    params.require(:destination_channel_id)
+
+    raise Discourse::InvalidAccess if !guardian.can_move_chat_messages?
+
+    @chat_channel = find_channel_and_check_access(params[:chat_channel_id])
+    @destination_channel = find_channel_and_check_access(params[:destination_channel_id])
+
+    message_ids = params[:message_ids].map(&:to_i)
+
+    begin
+      moved_messages = ChatMessageMoveService.new(
```

Comment:
We generally dont use the service terminology afaik.

Also the chat prefix is un necessary IMO as it's namespaced under DiscourseChat (I know other places do it in code currently but it will be dropped).

MoveMessageManager maybe?

Anyways feel free to keep service if you think that's the best fit here. Im just currently thinking of routes and naming in the plugin so thought I would raise this here.

Reply:
> We generally dont use the service terminology afaik.

I tried to come back with a rebuttal, but everything with `Service` in the name in core or in chat was made by me :laughing: 

> Also the chat prefix is un necessary IMO as it's namespaced under DiscourseChat

I actually don't have it namespaced. Will do that, and change to being called `DiscourseChat::MessageMover` (to echo `PostMover` in core)

Reply:
Or MessageMover as we have PostMover?

Reply:
Yes I suggested the same above :)

Reply:
Sorry was on mobile didn't see your answer :p